default.nix

```diff
@@ -15,42 +15,67 @@ rec {
     in
       pkg.offline_cache;
 
-  buildYarnPackageDeps = { name, packageJson, yarnLock, yarnNix ? null }:
+  buildYarnPackageDeps = { name, packageJson, yarnLock, yarnNix ? null, pkgConfig ? {} }:
     let
       yarnNix_ =
         if yarnNix == null then (generateYarnNix yarnLock) else yarnNix;
       offlineCache =
         loadOfflineCache yarnNix_;
+      moreBuildInputs = (pkgs.lib.flatten (builtins.map (key:
+        pkgConfig.${key} . buildInputs or []
+      ) (builtins.attrNames pkgConfig)));
+      postInstall = (builtins.map (key:
+        if (pkgConfig.${key} ? "postInstall") then
+          ''
+            test -e node_modules/${key} && (
+              cd node_modules/${key}
+              ${pkgConfig.${key}.postInstall}
+            )
+          ''
+        else
+          ""
+      ) (builtins.attrNames pkgConfig));
     in
     stdenv.mkDerivation {
       name = "${name}-modules";
 
       phases = ["buildPhase"];
-      buildInputs = [ yarn nodejs-6_x ];
+      buildInputs = [ yarn nodejs ] ++ moreBuildInputs;
```

Comment:
I don't know how to do it without forcing their evaluation. I'm happy if we keep that in mind and solve it only if it becomes a problem.